tool/planet/etcd.go

```diff
@@ -272,22 +273,33 @@ func etcdUpgrade(rollback bool) error {
 		}
 	}
 
-	// reset the kubernetes api server to take advantage of any new etcd settings that may have changed
-	// this only happens if the service is already running
-	status, err := getServiceStatus(APIServerServiceName)
-	if err != nil {
-		log.Warnf("Failed to query status of service %v. Continuing upgrade. Error: %v", APIServerServiceName, err)
-	} else {
-		if status != "inactive" {
-			tryResetService(ctx, APIServerServiceName)
-		}
-	}
-
 	log.Info("Upgrade complete")
 
 	return nil
 }
 
+// restartEtcdClients - because the etcd cluster has been recreated, all clients need to be refreshed so their
+// watches are not pointing at incorrect revisions.
+func restartEtcdClients(ctx context.Context) {
+	services := []string{APIServerServiceName, PlanetAgentServiceName, FlannelServiceName}
+
+	for _, service := range services {
+		// reset the kubernetes api server to take advantage of any new etcd settings that may have changed
+		// this only happens if the service is already running
+		status, err := getServiceStatus(service)
+		if err != nil {
+			log.WithFields(log.Fields{
+				log.ErrorKey: err,
+				"service":    service,
+			}).Warn("Failed to query service status.")
+			return
+		}
+		if status != "inactive" {
```

Comment:
@knisbet @a-palchikov Btw, I've just realized that this `!= "inactive"` logic probably applies to kube-apiserver only, in order to filter out inactive apiservers running on standby masters? It probably should not kick in for other services.

Reply:
it also applies to flannel, since flannel isn't necessarily enabled / running due to support for third party plugins and wormhole. And I don't see it doing harm to do the same thing for gravity-agent. If the process is inactive, why should the etcd upgrade touch it.